### ISSUE TITLE
Deleting custom span element results in JavaScript error

### ISSUE THREAD

**Issue by jswiderski** [ckeditor/ckeditor5#5781]
## 📝 Provide detailed reproduction steps (if any)

1. Unpack attached sample application (it contains very simple span plugin), run `npm install` and `npx webpack --mode development`.
2. Load the sample in browser and put the cursor at the end of text.
3. Start deleting text with backspace
Sample application: [ckeditor5-span.zip](https://github.com/ckeditor/ckeditor5/files/3880221/ckeditor5-span.zip)

### ✔️ Expected result

Grayed text with custom span should get deleted.

### ❌ Actual result

JavaScript error gets thrown - move-operation-node-into-itself: Trying to move a range of nodes into one of nodes from that range.

![obraz](https://user-images.githubusercontent.com/5314687/69439091-89439180-0d46-11ea-9c73-2964cf779124.png)

## 📃 Other details

* Browser: Any
* OS: Any
* CKEditor version: 15.0.0
* Installed CKEditor plugins: Custom Span plugin

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
The plugin code:

```js
function Foo( editor ) {
	const schema = editor.model.schema;

	schema.register( 'ab-span', {
		allowWhere: '$text',
		allowContentOf: '$block'
	} );

	const conversion = editor.conversion;

	conversion.elementToElement( {
		view: {
			name: 'span'
		},
		model: 'ab-span'
	} );
}
```

The `ab-span` definition is a bit weird because it's allowed where the text is, but it can contain block content inside. We do not support such scenarios yet. You can only have inline content on the same level.

The other thing is that this element should be marked with at least `isInline`, but the only configuration we support now is the combination of `isInline` and `isBlock`, just like in https://ckeditor.com/docs/ckeditor5/latest/framework/guides/tutorials/implementing-an-inline-widget.html.

In other words – the only inline elements we support right now are inline objects that are rendered to widgets. We do not support any other inline elements in the model. All other inline features that we have use either text attributes or markers.

**Comment by fjoalland** [comment]
I have the same issue, I can't remove the span. I can see this post didn't change that much, is it still a bug?

**Comment by Reinmar** [comment]
To quote myself:

> In other words – the only inline elements we support right now are inline objects that are rendered to widgets. We do not support any other inline elements in the model. All other inline features that we have use either text attributes or markers.

If you tried to do the same thing that's described in the issue, then you're trying to do something that's not supported by CKEditor 5. Inline elements should come from model attributes.
